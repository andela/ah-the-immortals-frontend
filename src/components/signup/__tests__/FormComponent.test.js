import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Form from '../FormComponent';

describe('<Form/>', () => {
  it('renders form component', () => {
    const mockOnsubmit = jest.fn();
    const mockOnchange = jest.fn();
    const wrapper = mount(
      <Form
        handleSubmit={mockOnsubmit}
        handleChange={mockOnchange}
        signupdata={{
          email: 'test@gmail.com',
          errors: {}
        }}
        errorShow={{ password: false }}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
