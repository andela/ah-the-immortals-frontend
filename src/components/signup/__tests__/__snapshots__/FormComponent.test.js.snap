// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders form component 1`] = `
<Form
  handleChange={[MockFunction]}
  handleSubmit={[MockFunction]}
>
  <form
    className="modal-form"
    onSubmit={[MockFunction]}
  >
    <input
      className="form-control"
      name="email"
      onChange={[MockFunction]}
      placeholder="Email"
      required={true}
      type="email"
    />
    <input
      className="form-control"
      name="username"
      onChange={[MockFunction]}
      placeholder="Username"
      required={true}
      type="text"
    />
    <input
      className="form-control"
      name="password"
      onChange={[MockFunction]}
      placeholder="Password"
      required={true}
      type="password"
    />
    <input
      className="form-control"
      name="password_confirm"
      onChange={[MockFunction]}
      placeholder="Confirm Password"
      required={true}
      type="password"
    />
    <input
      className="form-control btn"
      type="submit"
      value="SIGN UP"
    />
  </form>
</Form>
`;
